firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/Serial/Nonin/PacketReceiver.h

```diff
@@ -25,76 +25,26 @@
 
 #include "FrameBuffer.h"
 #include "FrameReceiver.h"
+#include "Types.h"
 
-namespace Pufferfish {
-namespace Driver {
-namespace Serial {
-namespace Nonin {
+namespace Pufferfish::Driver::Serial::Nonin {
 
-/* Enum class for amplitude representation of signal quality  */
-enum class SignalAmplitude {
-  no_perfusion = 0,  /// No loss in signal quality
-  red_perfusion,     /// Red Perfusion – Amplitude representation of low signal
-                     /// quality
-  yellow_perfusion,  /// YPRF: Yellow Perfusion – Amplitude representation of
-                     /// medium signal quality
-  green_perfusion    /// Green Perfusion – Amplitude representation of high signal
-                     /// quality
+enum class PacketInputStatus {
+  output_ready = 0,  /// Input is available to read output
+  waiting,           /// Input is wait to read more bytes
+  frame_loss         /// missed one or more frames in previous received packet
 };
-
-/* Structure defines the sensor data in packet for measurements */
-struct StatusByteStruct {
-  bool bit7;
-  bool sensor_disconnect;
-  bool artifact;
-  bool out_of_track;
-  bool sensor_alarm;
-  SignalAmplitude signal_perfusion;
-};
-
-/* Packet of 25 frames */
-static const size_t packet_size = 25;
-using Packet = std::array<Frame, packet_size>;
-/* Status Byte error of 25 frames */
-using StatusByteError = std::array<StatusByteStruct, packet_size>;
-/* PLETH for 25 frames */
-using Pleth = std::array<uint8_t, packet_size>;
-
-/* Structure defines the sensor data in packet for measurements */
-struct PacketMeasurements {
-  /* Heart Rate and SpO2 measurements */
-  uint16_t heart_rate;
-  uint8_t spo2;
-  uint8_t spo2_d;
-  uint8_t spo2_d_fast;
-  uint8_t spo2_d_beat;
-  uint16_t e_heart_rate;
-  uint8_t e_spo2;
-  uint8_t e_spo2_d;
-  uint16_t heart_rate_d;
-  uint16_t e_heart_rate_d;
-  uint8_t nonin_oem_revision;
-  /* PLETH measurements */
-  Pleth packet_pleth;
-  /* StatusByteErrors measurements */
-  std::array<bool, packet_size> bit7;
-  std::array<bool, packet_size> sensor_disconnect;
-  std::array<bool, packet_size> artifact;
-  std::array<bool, packet_size> out_of_track;
-  std::array<bool, packet_size> sensor_alarm;
-  std::array<SignalAmplitude, packet_size> signal_perfusion;
-};
-
-static const uint8_t mask_6bit = 0x7F;
+using PacketOutputStatus = Pufferfish::Driver::Serial::Nonin::FrameOutputStatus;
+static const uint8_t spo2_missing_data = 0x7F;
```

Comment:
Is `spo2_missing_data` a mask? If so, should `spo2_missing_data` be moved to be with the other masks in Types.h? Also, what does "spo2 missing data" mean? Would it be clear if we just named it to "spo2 missing"? I think "spo2 missing data" is too ambiguous because it could either be saying "There is something called SpO2 and it needs some data which is currently missing", "We are missing the data for SpO2", or "Here is some data describing why SpO2 is missing".

If `spo2_missing_data` is a non-mask value defined in the datasheet, perhaps we should move it to `Types.h` (maybe into a `SpecialConstants` or `ErrorConstants` or `Flags` internal namespace) and use the datasheet's format for representing the value, e.g. `static const uint8_t spo2_missing = 127; // Normally we'd use hex, but the datasheet lists it as 127 in decimal so we do the same thing here`.